### PR TITLE
feat: add gethash.py

### PR DESCRIPTION
**Pull request by lijie0123**
Why:
- add gethash.py to this repo
- it join old two scripts, get hash from github and gethash from fileserver
- it removes credential, so all old caller need to be refracted
- it is python3 version, because old caller need to be refracted anyway as above
- repo can be 'tikv/pd' to fully qualify

Todo:
- refract caller from `python gethash.py` to `environment{GHTOKEN=credential()} {python3 gethash.py}`

ref #2439

### CODE REVIEW

File: tools/gethash/Readme.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# How to use
+GHTOKEN=<your github token> ./gethash.py -repo tikv -version master
```
Comment (wuhuizuo) [inline tools/gethash/Readme.md:2]:
```suggestion
GH_TOKEN=<your github token> ./gethash.py -repo tikv -version master
```

File: tools/gethash/gethash.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python3
+
+from time import sleep
+import urllib.request as urllib2
```
Comment (wuhuizuo) [inline tools/gethash/gethash.py:4]: dose it need pip install?

File: tools/gethash/gethash.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+import argparse
+import json
+import re
+import os
+
+token = os.environ.get('GHTOKEN')
```
Comment (wuhuizuo) [inline tools/gethash/gethash.py:10]:
```suggestion
token = os.environ.get('GH_TOKEN')
```

File: tools/gethash/gethash.py (hunk 2 above)
Comment (lijie0123) [reply tools/gethash/gethash.py:4]: system libary

File: tools/gethash/gethash.py (hunk 3 above)
Comment (lijie0123) [reply tools/gethash/gethash.py:10]: some places has used already GHTOKEN https://github.com/PingCAP-QE/ci/blob/0997d6b809168c9cb136467a9b7b0dbb266d1f6c/jenkins/pipelines/cd/dev-build.groovy#L64

File: tools/gethash/gethash.Dockerfile [hunk 4]
```diff
@@ -0,0 +1,3 @@
+FROM pingcap/pingcap-base:v1
```
Comment (wuhuizuo) [inline tools/gethash/gethash.Dockerfile:1]: please make the file name simple: `Dockerfile`

File: tools/gethash/Readme.md (hunk 1 above)
Comment (wuhuizuo) [reply tools/gethash/Readme.md:2]: make it same with gh cli tool.

**Comment by wuhuizuo** [comment]
Maige label this

**Pull request merged**